local-rest-scorer/src/main/java/ai/h2o/mojos/deploy/local/rest/error/ModelsExceptionHandler.java

```diff
@@ -0,0 +1,48 @@
+package ai.h2o.mojos.deploy.local.rest.error;
+
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class ModelsExceptionHandler extends ResponseEntityExceptionHandler {
+  private static final Logger log = LoggerFactory.getLogger(ModelsExceptionHandler.class);
+
+  /**
+   * Custom Exception handler for ResponseStatusException type.
+   */
+  @ExceptionHandler(ResponseStatusException.class)
+  public ResponseEntity<Object> handleResponseStatusException(
+      ResponseStatusException exception, WebRequest request) {
+    log.error("Runtime exception occurred : {}", exception.getMessage(), exception);
+    return ResponseEntity
+        .status(exception.getStatus())
+        .body(ImmutableMap
+          .builder()
+          .put("detail", exception.getMessage())
+          .build()
+        );
+  }
+
+  /**
+   * Custom Exception handler for all Exception type.
+   */
+  @ExceptionHandler(Exception.class)
+  public ResponseEntity<Object> handleAllException(
+      Exception exception, WebRequest request) {
+    log.error("Unexpected exception occurred : {}", exception.getMessage(), exception);
+    return ResponseEntity
+        .status(HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
nit: why not `ResponseEntity.internalServerError()...`

Reply:
thx @achraf-mer , i m not aware there is a builder to construct the response entity, updated c7042eb79de6a6ac1ab4190a4c4791d4e9e39771